bundles/org.openhab.binding.elkm1/src/main/java/org/openhab/binding/elkm1/internal/elk/ElkAlarmConnection.java

```diff
@@ -0,0 +1,304 @@
+        this.config = config;
+        this.factory = factory;
+    }
+
+    /**
+     * Initializes the connection by connecting to the elk and verifying we get
+     * basic data back.
+     *
+     * @return true if successfully initialized.
+     */
+    public boolean initialize() {
+        if (config.useSSL) {
+            TrustManager[] trustAllCerts = new TrustManager[] { new X509TrustManager() {
+                @Override
+                public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                    return new X509Certificate[0];
+                }
+
+                @Override
+                public void checkClientTrusted(java.security.cert.X509Certificate[] certs, String authType) {
+                }
+
+                @Override
+                public void checkServerTrusted(java.security.cert.X509Certificate[] certs, String authType) {
+                }
+            } };
+
+            SSLContext sc;
+            try {
+                sc = SSLContext.getInstance("SSL");
+                sc.init(null, trustAllCerts, new java.security.SecureRandom());
+                sFactory = sc.getSocketFactory();
+            } catch (KeyManagementException | NoSuchAlgorithmException e) {
+                logger.error("An error has occured while creating the trust manager to connect to Elk alarm: {}:{}",
+                        config.ipAddress, config.port, e);
+            }
+
+        } else {
+            sFactory = SocketFactory.getDefault();
+        }
+
+        try {
+            socket = sFactory.createSocket(config.ipAddress, config.port);
+        } catch (IOException e) {
+            logger.error("Unable to open connection to Elk alarm: {}:{}", config.ipAddress, config.port, e);
+        }
+
+        if (config.useSSL) {
+            if (!sslLogin()) {
+                return false;
+            }
+        }
+
+        running = true;
+        elkAlarmThread = new Thread(new ReadingDataThread());
+        elkAlarmThread.start();
+
+        return socket != null && !socket.isClosed();
+    }
+
+    /**
+     * Called to login to the Elk using the given username and password.
+     *
+     * @return True if connection is established, false if it is not.
+     */
+    public boolean sslLogin() {
+        ((SSLSocket) socket).setEnabledProtocols(new String[] { "TLSv1" });
+        try {
+            try (BufferedWriter out = new BufferedWriter(
+                    new OutputStreamWriter(socket.getOutputStream(), StandardCharsets.UTF_8))) {
+                out.write(config.username + "\r\n");
+                out.write(config.password + "\r\n");
+                out.flush();
+            }
+
+            try (BufferedReader in = new BufferedReader(
+                    new InputStreamReader(socket.getInputStream(), StandardCharsets.UTF_8))) {
+
+                // Read back username and password
+                logger.debug("Elk Login Readback: ", in.readLine());
```

Comment:
You are still missing the "{}" in all of them. Should we maybe wrap those for lines by `if( logger.isDebugEnabled() ) {...}` to avoid the calls if not necessary.